Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -301,19 +318,19 @@ UPSSpotComponent* UPSWorldSubsystem::GetCurrentSpot() const
 	}
 
 	const FPlayerTag& PlayerTag = PlayerCharacter->GetPlayerTag();
-	if (PSSpotComponentArrayInternal.IsEmpty() || !PlayerTag.IsValid())
+	if (!PlayerTag.IsValid() || CurrentRowNameInternal.IsNone())
```

Comment:
`GetCurrentSpot` can singnificantly simplified now: no need for any logic in .cpp file at all.
Before: .h + .cpp
```
UFUNCTION(BlueprintCallable, BlueprintPure, Category="C++")
class UPSSpotComponent* GetCurrentSpot() const;
```
Now .h-only:
```
UFUNCTION(BlueprintCallable, BlueprintPure, Category="C++")
class FORCEINLINE UPSSpotComponent* GetCurrentSpot() const { return FindSpotByRowName(CurrentRowNameInternal); }
```

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/59/commits/8f5795e2e0f76f46a7ed92bafa5e548cd98ea431